plugins/systemd/systemd_mem

```diff
@@ -51,28 +58,51 @@ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SO
 
 . "$MUNIN_LIBDIR/plugins/plugin.sh"
 
+# at least on debian stretch services report this as unlimited / unknown value
+SYSTEMD_UNLIMITED="18446744073709551615"
+
+for_services() {
+    if [ "$all_services" != "" ]; then
+        systemctl --type=service --state=running --no-pager --no-legend --output=short-precise \
+          | awk '{sub(".service$", "", $1); print $1}'
+    else
+        echo ${services:-"munin-node"}
+    fi
+}
+
 output_config() {
     echo "graph_title Systemd Service Memory Usage"
     echo "graph_info Memory usage for selected services as reported by systemctl"
     echo "graph_vlabel bytes"
     echo "graph_args --base 1024 --lower-limit 0"
     echo "graph_category memory"
-    for service in ${services:-"munin-node"}; do
+    for service in $(for_services); do
         clean_name="$(clean_fieldname "$service")"
-        description=$(systemctl show "$service" --all | grep ^Description= | cut -d '=' -f 2)
+        description=$(systemctl show "$service" --property=Description | cut -d '=' -f 2)
```

Comment:
Maybe use `-f 2-` (instead of `-f 2`) - otherwise content including an equal character is truncated.

Reply:
this came from the original plugin. nonetheless `-f 2-` seems to be a better fit for description. for the memory numbers: as we do not expect an `=` to make an appearance in there i kept the `-f 2` for them.